ivc/src/poseidon_quadri/columns.rs

```diff
@@ -0,0 +1,74 @@
+/// The column layout will be as follow, supposing a state size of 3 elements:
+/// | C1 | C2 | C3 | C4  | C5  | C6  | ... | C_(k) | C_(k + 1) | C_(k + 2) |
+/// |--- |----|----|-----|-----|-----|-----|-------|-----------|-----------|
+/// |  x |  y | z  | x'  |  y' |  z' | ... |  x''  |     y''   |    z''    |
+///                | MDS \circ SBOX  |     |        MDS \circ SBOX         |
+///                |-----------------|     |-------------------------------|
+///                   Divided in 5
+///                 blocks of degree 2
+///                   constraints
+/// where (x', y', z') = MDS(x^7, y^7, z^7), i.e. the result of the linear layer
+/// We will have, for N full rounds:
+/// - `3` input columns
+/// - `5 N * 3` round columns, indexed by the round number and the index in the state,
+/// the number of rounds.
+/// The round constants are also added as columns.
+// IMPROVEME: should be split the SBOX and the MDS?
+// IMPROVEME: round constants should be public
+// IMPROVEME: round constants should be part of the expression framework. The
+// expression must be changed to support this.
+use kimchi_msm::columns::{Column, ColumnIndexer};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum PoseidonColumn<const STATE_SIZE: usize, const NB_FULL_ROUND: usize> {
+    Input(usize),
+    // state, nb round
```

Comment:
I think in the interpreter code your indexing is rather `(round, state_ix)`? (swapped)

Reply:
Done